eng/testing/tests.singlefile.targets

```
@@ -103,6 +107,24 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="RemoveDbgBinsFromTestR2ROutput"
+          Condition="'$(TestReadyToRun)' == 'true'"
+          BeforeTargets="_CopyFilesMarkedCopyLocal">
+    <ItemGroup>
+      <ReferenceCopyLocalPaths Remove="@(ReferenceCopyLocalPaths)"
+                               Condition="'%(ReferenceCopyLocalPaths.Extension)' == '.dbg'" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="RemoveDbgBinsFromTestR2RPublish"
+          Condition="'$(TestReadyToRun)' == 'true'"
+          BeforeTargets="_CopyResolvedFilesToPublishPreserveNewest">
+    <ItemGroup>
+      <_ResolvedFileToPublishPreserveNewest Remove="@(_ResolvedFileToPublishPreserveNewest)"
+                                            Condition="'%(_ResolvedFileToPublishPreserveNewest.Extension)' == '.dbg'" />
```

nit: use item functions for better perf and less verbose binlog logging:

```suggestion
      <_ResolvedFileToPublishPreserveNewest Remove="@(_ResolvedFileToPublishPreserveNewest->WithMetadataValue('Extension', '.dbg'))" />
```

---

Oh I didn't know those existed. Will change them now. Thanks for letting me know.